engine/src/test/java/io/zeebe/engine/processing/variable/mapping/MessageCatchElementOutputMappingTest.java

```diff
@@ -187,7 +187,7 @@ public void shouldMergeMessageVariablesByDefault() {
     // then
     final Record<VariableRecordValue> variableEvent =
         RecordingExporter.variableRecords(VariableIntent.CREATED)
-            .withProcessInstanceKey(processInstanceKey)
+            .withScopeKey(processInstanceKey)
```

Comment:
was this really necessary or is just making the test more focussed?

Reply:
It is necessary. Otherwise, the test would fail. Now, the variables are first stored locally in the scope of the element. So, it would pick these variables instead of the ones of the process instance. 

Previously, the variables were stored as temporary variables.